lib/gpu/gpuDeviceInterface.hpp

```diff
@@ -37,7 +37,7 @@ class gpuDeviceInterface : public kotekan::kotekanLogging {
      * NOTE: if accessing an existing named region then len must match the existing
      * length or the system will throw an assert.
      */
-    void* get_gpu_memory_array(const std::string& name, const uint32_t index, const uint32_t len);
+    void* get_gpu_memory_array(const std::string& name, const uint32_t index, const size_t len);
```

Comment:
Oh this is potentially even more painful... The `hsaDeviceInterface`, and `hipDeviceInterface` also need to be updated, which will likely cascade into more areas in those two frameworks.   It's a bit tricky to test that, because one only really works on the old ROCm drivers code on CHIME, and the other just isn't used outside of testing.  But at the very least we should update the signatures in those so we don't immediately fail in those builds.

Reply:
Okay, so I tried propagating this all through cl/hsa/hip directories at the level of grepping for things.